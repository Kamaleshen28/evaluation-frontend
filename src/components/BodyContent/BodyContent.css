.body-content-wrapper {
  padding: 100px 240px 100px 240px;
}

.all-event-card-container {
  display: flex;
  flex-wrap: wrap;
  gap: 20px
}