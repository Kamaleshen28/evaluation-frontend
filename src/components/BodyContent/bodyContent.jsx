import React, { useEffect, useState } from 'react';
import './BodyContent.css';
import axios from 'axios';
import Card from '../Card/card';
import FilterComponent from '../FilterComponent/filterComponent';


export default function BodyContent() {

  const [formData, setFormData] = React.useState(
    {
      filter: ''
    });

  // console.log('IN BODYCONTEXT', formData);

  // ---------------
  const [searchInput, setSearchInput] = useState('');
  const [storedData, setStoredData] = useState([]);


  const [eventData, setEventData] = useState([]);
  // const navigate = useNavigate();

  const fetchData = async () => {
    const data = await axios.get('http://localhost:8000/api/events', {
      headers: {
      }
    });

    (data.data).sort(function (a, b) {

      return new Date(a.datetime).getTime() - new Date(b.datetime).getTime();
    });

    setEventData(data.data);
    setStoredData(data.data);
    console.log('fetchOVER ');

  };
  useEffect(() => {
    fetchData();
  }, []);

  const upadteEventDataState = () => {
    if (formData.filter === 'ALL') {
      setEventData(storedData);
    } else if (formData.filter === 'BOOKMARKED') {
      setEventData(filterBookmakedData());
    } else if (formData.filter === 'REGISTERED') {
      console.log('Its REgu');
      setEventData(filterRegisteredEvent());
    } else if (formData.filter === 'SEATS AVAILABLE') {
      console.log('Its seat');
      setEventData(filterSeatsAvailableEvent());
    }
  };

  const handleOnClickBookmark = async (id, updatedValue) => {
    console.log('UPbooooo: ', updatedValue);
    const data = { 'isBookmarked': updatedValue };
    const config = {
      method: 'patch',
      url: `http://localhost:8000/api/events/${id}`,
      headers: {},
      data: data
    };
    const response = await axios(config);
    console.log('response: ', response);
    await fetchData();
    console.log('FetchDONE: ');
    upadteEventDataState();

    console.log('BOOOOOOKMARK: ,', response);

  };
  //-----------------------------

  const upadteEventDataStateForRegisteration = (id, updatedValue) => {
    // const upadtedData =
    setEventData(eventData.map(eachEventData => {
      if (eachEventData.id === id) {
        return { ...eachEventData, isRegistered: updatedValue };
      } else {
        return { ...eachEventData };
      }
    }));
  };


  const handleRegisterButtonClick = async (id, updatedValue) => {
    console.log('UP: ', updatedValue);
    const data = { 'isRegistered': updatedValue };
    const config = {
      method: 'patch',
      url: `http://localhost:8000/api/events/${id}`,
      headers: {},
      data: data
    };
    const response = await axios(config);
    upadteEventDataStateForRegisteration(id, updatedValue);

    console.log('BOOOOOOKMARK: ,', response);

  };
  // --------------------------------

  const handleSearchBoxChange = (event) => {
    setSearchInput(event.target.value);
    if (event.target.value.trim() === '') {
      upadteEventDataState();
    } else {
      const filteredFoodData = storedData.filter(eachFood => ((eachFood.name).startsWith(event.target.value.trim())));
      setEventData(filteredFoodData);
    }
  };

  const filterBookmakedData = () => {
    return storedData.filter(eachEventData => {
      if (eachEventData.isBookmarked) {
        return eachEventData;
      }
    });
  };
  const filterRegisteredEvent = () => {
    return storedData.filter(eachEventData => {
      if (eachEventData.isRegistered) {
        return eachEventData;
      }
    });
  };
  const filterSeatsAvailableEvent = () => {
    return storedData.filter(eachEventData => {
      if (eachEventData.areSeatsAvailable) {
        return eachEventData;
      }
    });
  };
  //---------------------------------
  const handleRadioButtonChange = (event) => {
    const { name, value, type, checked } = event.target;
    // console.log('VALUE: ', value);
    if (value === 'ALL') {
      setEventData(storedData);
    } else if (value === 'BOOKMARKED') {
      console.log('Its boooooooo');
      setEventData(filterBookmakedData());
    } else if (value === 'REGISTERED') {
      console.log('Its REgu');
      setEventData(filterRegisteredEvent());
    } else if (value === 'SEATS AVAILABLE') {
      console.log('Its seat');
      setEventData(filterSeatsAvailableEvent());
    }


    setFormData(prevFormData => {
      return {
        ...prevFormData,
        [name]: type === 'checkbox' ? checked : value
      };
    });



  };

  const renderAllEvents = eventData.map(eachEventData => {
    return (< Card
      key={eachEventData.id}
      {...eachEventData}
      handleOnClickBookmark={handleOnClickBookmark}
      handleRegisterButtonClick={handleRegisterButtonClick}
    />);
  });

  return (
    <div className="body-content">
      <div className="body-content-wrapper">
        <div className="filter-component-wrapper">
          <FilterComponent
            searchInput={searchInput}
            handleSearchBoxChange={handleSearchBoxChange}
            handleRadioButtonChange={handleRadioButtonChange}
            filter={formData.filter}
          />
        </div>
        <div className="all-event-card-container">
          {renderAllEvents}

        </div>
      </div>
    </div>
  );
}
