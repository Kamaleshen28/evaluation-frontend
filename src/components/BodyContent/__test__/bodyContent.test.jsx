import { render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import BodyContent from '../bodyContent';
import axios from 'axios';

jest.mock('react-router-dom');

const mockEventData =
{
  data: [{
    'id': 1,
    'name': 'Battle of the Bands',
    'description': 'Get ready for Battle of the Bands, where the hottest up-and-coming rock groups will compete for the ultimate prize. With heart-pumping beats and electrifying performances, you won\'t want to miss this adrenaline-fueled event. Each band will bring their A-game, leaving everything on the stage for the chance to be crowned champion. Don\'t miss out on the most unforgettable rock competition of the year!',
    'venue': 'All Stars Arena, Las Vegas, NV, USA',
    'datetime': '2023-03-01T05:00:00.000Z',
    'timezone': 'America/Los_Angeles',
    'areSeatsAvailable': true,
    'isRegistered': true,
    'isBookmarked': true,
    'imgUrl': 'https://i.ibb.co/3zbdvWX/battle-of-bands.jpg'
  }]
};


describe('card', () => {
  it('should render filterComponent correctly', async () => {
    jest.spyOn(axios, 'get').mockResolvedValue(mockEventData);
    const { asFragment } = render(<BodyContent />);
    await waitFor(() => {
      expect(screen.getAllByText('All Stars Arena, Las Vegas, NV, USA')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});



