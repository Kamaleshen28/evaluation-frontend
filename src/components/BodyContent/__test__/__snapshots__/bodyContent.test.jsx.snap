// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should render filterComponent correctly 1`] = `
<DocumentFragment>
  <div
    class="body-content"
  >
    <div
      class="body-content-wrapper"
    >
      <div
        class="filter-component-wrapper"
      >
        <div
          class="body-header"
        >
          <form>
            <legend>
              FILTER
            </legend>
            <input
              id="ALL"
              name="filter"
              type="radio"
              value="ALL"
            />
            <label
              for="ALL"
            >
              ALL
            </label>
            <br />
            <input
              id="BOOKMARKED"
              name="filter"
              type="radio"
              value="BOOKMARKED"
            />
            <label
              for="BOOKMARKED"
            >
              BOOKMARKED
            </label>
            <br />
            <input
              id="REGISTERED"
              name="filter"
              type="radio"
              value="REGISTERED"
            />
            <label
              for="REGISTERED"
            >
              REGISTERED
            </label>
            <br />
            <input
              id="SEATS AVAILABLE"
              name="filter"
              type="radio"
              value="SEATS AVAILABLE"
            />
            <label
              for="SEATS AVAILABLE"
            >
              SEATS AVAILABLE
            </label>
            <br />
            <br />
          </form>
          <div
            class="searchbar-lens"
          >
            <input
              class="search-box"
              name="input"
              placeholder="EVENT NAME"
              value=""
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-magnifying-glass search-button"
              data-icon="magnifying-glass"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="all-event-card-container"
      >
        <div
          class="card"
        >
          <div
            class="card-wrapper"
          >
            <div
              class="card-top-section"
            >
              <img
                alt=""
                class="card-top-section-image"
                data-testid="clickable-area"
                src="https://i.ibb.co/3zbdvWX/battle-of-bands.jpg"
              />
            </div>
            <div
              class="card-middle-section"
            >
              <span
                class="card-event-name"
              >
                Battle of the Bands
              </span>
              <span
                class="card-event-description"
              >
                Get ready for Battle of the Bands, where the hottest up-and-coming rock groups will compete for the ultimate prize. With heart-pumping beats and electrifying performances, you won't want to miss this adrenaline-fueled event. Each band will bring their A-game, leaving everything on the stage for the chance to be crowned champion. Don't miss out on the most unforgettable rock competition of the year!
              </span>
              <span
                class="full-card-evnt-venue"
              >
                <span
                  class="venue-title"
                >
                  VENUE:
                </span>
                 All Stars Arena, Las Vegas, NV, USA
              </span>
              <span
                class="full-card-evnt-venue"
              >
                <span
                  class="date-title"
                >
                  DATE:
                </span>
                 2023-03-01T05:00:00.000Z
              </span>
            </div>
            <div
              class="card-bottom-section"
            >
              <div
                class="card-bottom-upper"
              >
                <div
                  class="card-bottom-upper-left"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle-check tick-icon"
                    data-icon="circle-check"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="registration-status-text"
                  >
                    REGISTERED
                  </span>
                </div>
                <div
                  class="card-bottom-upper-right"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-bookmark "
                    data-icon="bookmark"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 48V487.7C0 501.1 10.9 512 24.3 512c5 0 9.9-1.5 14-4.4L192 400 345.7 507.6c4.1 2.9 9 4.4 14 4.4c13.4 0 24.3-10.9 24.3-24.3V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
